clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/AddExecutor.scala

```diff
@@ -42,8 +46,142 @@ class AddExecutor[TI <: TransferIndex](addCommand: AddCommand[TI])
       Future
         .failed(_) logErrorMsg s"Metadata at $location cannot be added to Clio", {
         decoded =>
-          webClient.upsert(addCommand.index)(addCommand.key, decoded)
+          {
+            for {
+              existingMetadata <- if (!addCommand.forceUpdate) {
+                queryForKey(webClient).logErrorMsg(
+                  s"Could not query the $name. No files have been added."
+                )
+              } else {
+                Future.successful(None)
+              }
+              upsertResponse <- addFiles(
+                webClient,
+                decoded,
+                existingMetadata
+              ).fold(
+                left => Future.failed(left).logErrorMsg(left.getMessage),
```

Comment:
Is the `logErrorMsg` redundant? It'll wrap `left` with a new `RuntimeException`, with the same message, so we'll see it twice in the ending stack trace.